### PR TITLE
Fix S1118 FN: empty primary constructor

### LINKED ISSUE #8122: Fix S1118 FN: empty primary constructor
The rule does not take into account when the class is defining an empty primary constructor.

```cs
﻿public class PrimaryCtor() // FN
{
    public static string Concatenate(string s1, string s2)
    {
        return s1 + s2;
    }

    public static string Prop { get; set; }
}
```

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes #8122

Blocked by https://github.com/SonarSource/sonar-dotnet/pull/8209

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/ClassWithOnlyStaticMember.cs [hunk 1]
```diff
@@ -79,18 +80,26 @@ private static void CheckConstructors(SonarSymbolReportingContext context, IName
...
                 var syntaxReferences = constructor.DeclaringSyntaxReferences;
                 foreach (var syntaxReference in syntaxReferences)
                 {
-                    if (syntaxReference.GetSyntax() is ConstructorDeclarationSyntax constructorDeclaration)
+                    switch (syntaxReference.GetSyntax())
                     {
-                        context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), reportMessage));
+                        case ConstructorDeclarationSyntax constructorDeclaration:
+                            var reportMessage = string.Format(MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected);
+                            context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), reportMessage));
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/ClassWithOnlyStaticMember.cs:94]: Educational: isn't Diagnostic.Create() doing the `string.Format` already?
`context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected));`

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/src/SonarAnalyzer.CSharp/Rules/ClassWithOnlyStaticMember.cs [hunk 2]
```diff
@@ -79,18 +80,26 @@ private static void CheckConstructors(SonarSymbolReportingContext context, IName
...
+                            break;
+                        case ClassDeclarationSyntax classDeclaration when ClassDeclarationSyntaxWrapper.IsInstance(classDeclaration)
+                                && (ClassDeclarationSyntaxWrapper)classDeclaration is var classDeclarationWrapper
+                                && classDeclarationWrapper.ParameterList is { Parameters.Count: 0 }:
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/ClassWithOnlyStaticMember.cs:98]: Waiting for the `ParameterList()` extension method from #8209 to simplify this expression
https://github.com/SonarSource/sonar-dotnet/pull/8209/files#diff-0aed6645a137b426dc5eb96dcf45438091d2b6a5dba073c7c688407106ab9f35R54

**Comment by martin-strecker-sonarsource** [comment]
Peach validation: No changes

**Pull request merged**